templates/addons/calico/patches/remove-aws-reference.yaml

```diff
@@ -0,0 +1,5 @@
+# Drops the FELIX_AWSSRCDSTCHECK field
+# json patch doesn't allow for dropping fields in array by name
+# https://github.com/projectcalico/calico/issues/5101
+- op: remove
+  path: "/spec/template/spec/containers/0/env/12"
```

Comment:
Is there any way to identify this by the key (the name) rather than the array index number? This is a pretty unreliable contract IMO.

Reply:
no [json patch](https://datatracker.ietf.org/doc/html/rfc6902#appendix-A.4) doesn't allow by name, only by index.  

> The alternative is to copy all the variables and replace the env array completely. As we plan to upgrade very quickly to the calico 3.22 release which doesn't have this issue and this is a quick fix to unblock conformance tests but happy to go other way.

It's not ideal but I also don't see this changing often.  I am happy to switch to coping all the envs to the patch and doing a full replace.

Reply:
Comments look good, and since this is pinned to a specific calico spec, this will work for the time being.